### PR TITLE
JFB 6 

### PR DESCRIPTION
**Pull request by ya-kool**
1. ArrayList - is ObjectArray;
2. HashSet - is made by two variants (with map(ObjectSetAsMap) and with array of objects(ObjectSetAsArray)). Mark me which one i need to save(the other i'll drop as redundant)

### CODE REVIEW

File: src/main/java/by/yahor_kulesh/utils/ObjectSetAsArray.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    public ObjectSetAsArray() {
+       objects = new Object[16];
+    }
+
+    public boolean add(Object o) {
+        if(contains(o)) {
+            return false;
+        } else {
+            if(firstEmptyElement >= objects.length*0.75) {
```
Comment (AtticSiol) [inline src/main/java/by/yahor_kulesh/utils/ObjectSetAsArray.java]: define 0.75 as a constant

File: src/main/java/by/yahor_kulesh/utils/ObjectSetAsMap.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package by.yahor_kulesh.utils;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.function.Consumer;
+
+public class ObjectSetAsMap {
```
Comment (AtticSiol) [inline src/main/java/by/yahor_kulesh/utils/ObjectSetAsMap.java]: ObjectSetAsArray is enough, you can remove this one (ObjectSetAsMap) as redundant

File: src/main/java/by/yahor_kulesh/utils/ObjectSetAsArray.java (hunk 1 above)
Comment (ya-kool) [reply src/main/java/by/yahor_kulesh/utils/ObjectSetAsArray.java]: solved

File: src/main/java/by/yahor_kulesh/utils/ObjectSetAsMap.java (hunk 2 above)
Comment (ya-kool) [reply src/main/java/by/yahor_kulesh/utils/ObjectSetAsMap.java]: dropped

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**